crates/core/src/deps/system_contracts.rs

```diff
@@ -92,7 +97,19 @@ pub fn bytecode_from_slice(artifact_name: &str, contents: &[u8]) -> Vec<u8> {
 }
 
 pub fn load_builtin_contract(protocol_version: ProtocolVersionId, artifact_name: &str) -> Vec<u8> {
-    let artifact_path = format!("{artifact_name}.json");
+    let artifact_path = format!(
+        "{}.json",
+        if artifact_name == "proved_batch"
+            || artifact_name == "proved_batch_impersonating"
+            || artifact_name == "playground_batch"
+            || artifact_name == "fee_estimate"
+            || artifact_name == "fee_estimate_impersonating"
+        {
+            "Bootloader"
+        } else {
+            artifact_name
+        }
+    );
```

Comment:
I think this logic breaks V28 and below. Could be why tests are currently failing.